contrib/aws/aws-sdk-go/aws/aws_test.go

```diff
@@ -75,3 +76,64 @@ func TestAWS(t *testing.T) {
 		assert.Equal(t, "http://ec2.us-west-2.amazonaws.com/", s.Tag(ext.HTTPURL))
 	})
 }
+
+func TestAnalyticsSettings(t *testing.T) {
+	cfg := aws.NewConfig().
+		WithRegion("us-west-2").
+		WithDisableSSL(true).
+		WithCredentials(credentials.AnonymousCredentials)
+
+	session := session.Must(session.NewSession(cfg))
+	assertRate := func(t *testing.T, mt mocktracer.Tracer, rate interface{}, opts ...Option) {
+		ws := WrapSession(session, opts...)
+		ec2.New(ws).DescribeInstancesWithContext(context.TODO(), &ec2.DescribeInstancesInput{})
+		spans := mt.FinishedSpans()
+		assert.Len(t, spans, 1)
+		s := spans[0]
+		assert.Equal(t, rate, s.Tag(ext.EventSampleRate))
+	}
+
+	t.Run("defaults", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		assertRate(t, mt, nil)
+	})
+
+	t.Run("global", func(t *testing.T) {
+		t.Skip("global flag disabled")
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		rate := globalconfig.AnalyticsRate()
+		defer globalconfig.SetAnalyticsRate(rate)
```

Comment:
It would but I think that's a bit too yucky imho. I prefer the simpler, more explicit and easier to read method. Generally if code needs inline comments its an indication of something not easy to understand :)

Reply:
However I have had test setup functions in the past which returned a teardown function, which were used as:
```go
defer setupTeardown()()
```
That's a bit shorter but still slightly confusing 😄